src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -514,6 +516,51 @@ static bool slash_boot_exists(void) {
         return (cache = false);
 }
 
+static bool fstab_is_same_device(const char *where, const char *what) {
+        _cleanup_endmntent_ FILE *f = NULL;
+        _cleanup_free_ char *dev = NULL;
+        struct mntent *me;
+        int r;
+
+        assert(where);
+        assert(what);
+
+        f = setmntent(fstab_path(), "re");
+        if (!f) {
+                assert(errno != ENOENT); /* We already know the path has an fstab entry, so fstab must exist */
```

Comment:
Please return `false`. The file may be accidentally removed after the last read.